apps/teams-test-app/src/components/TeamsCoreAPIs.tsx

```diff
@@ -1,255 +1,88 @@
-    teamsCore.registerBeforeUnloadHandler((readyToUnload): boolean => {
-      setTimeout(() => {
-        readyToUnload();
-      }, delay);
-      alert(`beforeUnload received; calling readyToUnload in ${delay / 1000} seconds`);
-      setRegisterBeforeUnloadRes('Success');
-      return true;
-    });
-  };
-
-  const addStates = (): void => {
-    const newNumStates = totalStates + 1;
-    setTotalStates(newNumStates);
-    window.history.pushState({ some: 'state', id: newNumStates }, 'tab state' + newNumStates, '/testTab');
-    setAddStatesRes('total States: ' + newNumStates);
-    window.addEventListener(
-      'popstate',
-      (event): void => {
-        if (event.state && event.state.id) {
-          setAddStatesRes('onpopstate: back button clicked. total remaining state: ' + event.state.id);
+import { ApiWithoutInput, ApiWithTextInput } from './utils';
+
+const EnablePrintCapability = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'enablePrintCapability',
+    title: 'Enable Print Capability',
+    onClick: async () => {
+      teamsCore.enablePrintCapability();
+      return 'called';
+    },
+  });
+
+const Print = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'print',
+    title: 'Print',
+    onClick: async () => {
+      teamsCore.print();
+      return 'called';
+    },
+  });
+
+const RegisterOnLoadHandler = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'registerOnLoadHandler',
+    title: 'Register On Load Handler',
+    onClick: async setResult => {
+      teamsCore.registerOnLoadHandler((context: LoadContext): void => {
+        setResult('successfully called with context:' + JSON.stringify(context));
+      });
+
+      return 'registered';
+    },
+  });
+
+const RegisterBeforeUnloadHandler = (): React.ReactElement =>
+  ApiWithTextInput<number>({
+    name: 'registerBeforeUnload',
+    title: 'Register Before Unload',
+    onClick: {
+      validateInput: input => {
+        if (typeof input !== 'number') {
+          throw new Error('input should be a number');
         }
       },
-      false,
-    );
-  };
-
-  const returnRegisterBackButtonHandler = (): void => {
-    setRegisterBackButtonHandlerRes('total States: ' + totalStates);
-    pages.backStack.registerBackButtonHandler((): boolean => {
-      if (totalStates > 0) {
-        const newNumStates = totalStates - 1;
-        setTotalStates(newNumStates);
-        setRegisterBackButtonHandlerRes('back button clicked. total remaining state: ' + newNumStates);
+      submit: async (delay, setResult) => {
+        teamsCore.registerBeforeUnloadHandler((readyToUnload): boolean => {
+          setTimeout(() => {
+            readyToUnload();
+          }, delay);
+          alert(`beforeUnload received; calling readyToUnload in ${delay / 1000} seconds`);
+          setResult('Success');
+          return true;
+        });
+
+        return 'registered';
```

Comment:
Similar to my other comment, since this registration has a function passed in for the API to call, I think this might want to return a noHostSdkMsg